run_tests.py

```diff
@@ -76,6 +76,13 @@
 os.environ['PYTHONWARNDEFAULTENCODING'] = '1'
 # work around https://bugs.python.org/issue34624
 os.environ['MESON_RUNNING_IN_PROJECT_TESTS'] = '1'
+# python 3.11 adds a warning that in 3.15, UTF-8 mode will be default.
+# This is fantastic news, we'd love that. Less fantastic: this warning is silly,
+# we *want* these checks to be affected. Plus, the recommended alternative API
+# would (in addition to warning people when UTF-8 mode removed the problem) also
+# require using a minimum python version of 3.11 (in which the warning was added)
+# or add verbose if/else soup.
+warnings.filterwarnings('ignore', message="UTF-8 Mode affects .*getpreferredencoding", category=EncodingWarning)
```

Comment:
I think an `import warnings` is missing.

Reply:
Actually I used to have that, but then it went missing during a rebase. :D

Will have a chance to fix it soon.